### ISSUE TITLE
builtin.filecontent provider reports incorrect line numbers

### ISSUE THREAD

**Issue by tsanders-rh** [konveyor/analyzer-lsp#924]
# builtin.filecontent provider reports incorrect line numbers

## Summary
The `builtin.filecontent` provider is reporting incorrect line numbers for pattern matches. In the case below, it reports line **15573** for a file that only has **114 lines total**.

## Environment
- **Kantra version**: v0.8.1-alpha.1
- **SHA**: 8a6369de1247b304796433733e66eb55e07cd188
- **Rule**: `localhost-http-00001` from `cloud-readiness` ruleset
- **Provider**: `builtin.filecontent`

## Rule Definition
```yaml
- category: mandatory
  description: Local HTTP Calls
  effort: 7
  labels:
  - konveyor.io/target=cloud-readiness
  - konveyor.io/source
  - localhost
  message: The app is trying to access local resource by HTTP, please try to migrate the resource to cloud
  ruleID: localhost-http-00001
  when:
    builtin.filecontent:
      filePattern: .*\.(java|properties|jsp|jspf|tag|xml|txt|yaml)
      pattern: http(s)?://((localhost)|(127\.0\.0\.1))+(:[0-9]+)?(/.*)?
```

## Steps to Reproduce
1. Analyze the DayTrader sample application:
   ```bash
   git clone https://github.com/WASdev/sample.daytrader7.git
   kantra analyze \
       --input sample.daytrader7 \
       --output analysis-output \
       --target cloud-readiness
   ```

2. Check the output for `localhost-http-00001` violations in `output.yaml`

## Expected Behavior
The file `daytrader-ee7-web/src/main/java/com/ibm/websphere/samples/daytrader/web/prims/PingServlet2PDF.java` contains:

```java
// Line 85 (actual location of the localhost HTTP call)
String fileURL = "http://localhost:9080/daytrader/WAS_V7_64-bit_performance.pdf";
```

The file has only **114 lines total**.

Expected line number in output: **85**

## Actual Behavior
The analyzer output reports:

```yaml
localhost-http-00001:
  description: Local HTTP Calls
  category: mandatory
  incidents:
  - uri: file:///path/to/daytrader-ee7-web/src/main/java/com/ibm/websphere/samples/daytrader/web/prims/PingServlet2PDF.java
    message: The app is trying to access local resource by HTTP, please try to migrate the resource to cloud
    lineNumber: 15573  # ← WRONG! File only has 114 lines
    variables:
      matchingText: ""
  effort: 7
```

Actual line number in output: **15573** (impossible - file only has 114 lines)

## Verification
```bash
# File only has 114 lines
$ wc -l daytrader-ee7-web/src/main/java/com/ibm/websphere/samples/daytrader/web/prims/PingServlet2PDF.java
114 daytrader-ee7-web/src/main/java/com/ibm/websphere/samples/daytrader/web/prims/PingServlet2PDF.java

# Line 85 contains the actual localhost HTTP call
$ sed -n '85p' daytrader-ee7-web/src/main/java/com/ibm/websphere/samples/daytrader/web/prims/PingServlet2PDF.java
            String fileURL = "http://localhost:9080/daytrader/WAS_V7_64-bit_performance.pdf";
```

## Additional Context
This issue makes it difficult to use the analyzer output for automated test generation, as the line numbers cannot be trusted when using `builtin.filecontent` provider.

The line number 15573 appears to be either:
- A byte offset rather than a line number
- An accumulated count across multiple files
- A calculation error in the builtin provider

## Analysis Log Observations
The analysis log shows multiple `"unable to get code location"` errors which may be related:
```
level=error msg="unable to get code location" error="<nil>"
```

## Impact
- Incorrect test expectations in go-konveyor-tests
- Manual correction required for all builtin.filecontent violations
- Reduced trust in analyzer output accuracy

**Comment by eemcmullan** [comment]
@tsanders-rh Could you please confirm your OS? The `builtin.filecontent` condition uses different commands for Linux, Mac, and Windows.

**Comment by tsanders-rh** [comment]
@eemcmullan - I have verified with https://github.com/konveyor/kantra/releases/tag/v0.8.1-alpha.3.  All is resolved.   TY.
